### PR TITLE
Add MCUboot support for RV32M1_Vega

### PR DESCRIPTION
**Pull request by lylezhu2014**
1. Enable flash driver for RV32M1 SOC.
2. Add MCUboot support for RV32M1_Vega

### CODE REVIEW

File: dts/riscv32/rv32m1.dtsi [hunk 1]
```diff
@@ -289,17 +289,17 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
-			m4_flash: flash@0 {
+			flash0: flash@0 {
```
Comment (MaureenHelm) [inline dts/riscv32/rv32m1.dtsi]: Why are you changing this? Even though we're not using the cortex m cores, the hardware reference manual refers to these regions as m4 and m0 flashes.

File: boards/riscv32/rv32m1_vega/rv32m1_vega_ri5cy_defconfig [hunk 2]
```diff
@@ -12,3 +12,4 @@ CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_MULTI_LEVEL_INTERRUPTS=y
+CONFIG_FLASH=y
```
Comment (MaureenHelm) [inline boards/riscv32/rv32m1_vega/rv32m1_vega_ri5cy_defconfig]: This should not be in the default board config; it should get set in an application config.

File: soc/riscv32/openisa_rv32m1/dts_fixup.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+/* SoC level DTS fixup file */
+
+#if defined(CONFIG_SOC_OPENISA_RV32M1_RISCV32)
+
+#define DT_FLASH_DEV_BASE_ADDRESS		\
+				DT_OPENISA_RV32M1_FTFE_40023000_BASE_ADDRESS
+#define DT_FLASH_DEV_NAME				\
+				DT_OPENISA_RV32M1_FTFE_40023000_LABEL
```
Comment (MaureenHelm) [inline soc/riscv32/openisa_rv32m1/dts_fixup.h]: Long lines are ok for dts fixups.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/riscv32/rv32m1_vega/rv32m1_vega_ri5cy_defconfig (hunk 2 above)
Comment (lylezhu2014) [reply boards/riscv32/rv32m1_vega/rv32m1_vega_ri5cy_defconfig]: Done

File: dts/riscv32/rv32m1.dtsi (hunk 1 above)
Comment (lylezhu2014) [reply dts/riscv32/rv32m1.dtsi]: I removed it temporarily. And then I will try to raise a requirement to update the RM to make the flash region name easy to understand.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/flash_controller/openisa,rv32m1-ftfe.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+title: NXP Kinetis Flash Memory Module (FTFE)
```
[the remark this answers is not in the document]
Comment (lylezhu2014) [reply dts/bindings/flash_controller/openisa,rv32m1-ftfe.yaml]: Done

File: dts/bindings/flash_controller/openisa,rv32m1-ftfe.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+version: 0.1
+
+description: >
+    This binding gives for the NXP Kinetis Flash Memory Module E (FTFE)
```
[the remark this answers is not in the document]
Comment (lylezhu2014) [reply dts/bindings/flash_controller/openisa,rv32m1-ftfe.yaml]: Done

**Review by henrikbrixandersen — APPROVED** [review]

File: dts/riscv32/rv32m1.dtsi (hunk 1 above)
Comment (dleach02) [reply dts/riscv32/rv32m1.dtsi]: That was my doing @MaureenHelm . In my version of this change I didn't adopt the cortex naming and discussed with Lyle about changing the name.

File: dts/riscv32/rv32m1.dtsi [hunk 6]
```diff
@@ -282,7 +282,7 @@
 		};
 
 		flash-controller@40023000 {
-			compatible = "nxp,kinetis-ftfe";
+			compatible = "openisa,rv32m1-ftfe";
```
Comment (dleach02) [inline dts/riscv32/rv32m1.dtsi]: I would follow the normal naming with:

fctrl0: flash-controller@40023000

File: soc/riscv32/openisa_rv32m1/dts_fixup.h (hunk 3 above)
Comment (lylezhu2014) [reply soc/riscv32/openisa_rv32m1/dts_fixup.h]: OK, updated.

File: dts/riscv32/rv32m1.dtsi (hunk 6 above)
Comment (lylezhu2014) [reply dts/riscv32/rv32m1.dtsi]: This issue is not included by current pull request. It is better to create a new one for this issue.

**Review by MaureenHelm — APPROVED** [review]
Please rebase to fix the merge conflict

**Pull request merged**